doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+* In order to accommodate loss-free rescaling of unit-bearing quantities, real and integer literals in arithmetic contexts should have an internal representation that maintains a normalized significand and a power of ten scale.
+
+
+### String literals
+<a id="string-literals"/>
+
+String literals encode arbitrary character sequences used to specify and identify interface and species names.
+
+> ***string-literal*** ::= `"` ( [^&#92;"]* | _escape-sequence_ )\* `"`
+>
+> _escape-sequence_ ::= `\\` | `\"`
+
+#### Token value
+
+The value is taken as the characters between the initial and final `"` characters, followed by the replacement of each _escape-sequence_: `\\` by `\` and `\"` by `"`.
+
+#### Notes
+
+The value of a ***string-literal*** token does _not_ undergo NFKC normalization, and so comprises an arbitrary sequence of Unicode characters in NFC normalization. While the syntax allows arbitary white space within a literal, including line separators and similar, the interpreting environment might impose further restrictions on what constitutes a valid interface name, for example.
+
+
+### Punctuation
+<a id="punctuation"/>
+
+> _punctuation_ ::=
+>     ***plus-sign*** | ***minus-sign*** | ***multiplication-dot*** | ***division-slash*** | ***exponent-op*** | ***preferential-union*** |
+>     ***compare-equal*** | ***compare-not-equal*** | ***compare-less*** | ***compare-less-equal*** | ***compare-greater*** | ***compare-greater-equal*** |
+>     ***left-arrow*** | ***right-arrow*** | ***right-left-arrow*** | ***empty-set*** | ***square-root*** |
+>     ***assign-equal*** | ***semicolon*** | ***left-paren*** | ***right-paren*** | ***left-brace*** | ***right-brace*** |
+>     ***colon*** | ***bar*** | ***comma*** | ***period***
+
+Punctuation token definitions:
+
+> ***plus-sign*** ::= `+`
+>
+> ***minus-sign*** ::= U+002D `-` HYPHEN-MINUS | U+2212 `−`MINUS SIGN
+>
+> ***multiplication-dot*** ::= U+00B7 `·` MIDDLE DOT | U+22C5 `⋅` DOT OPERATOR
+>
+> ***asterisk*** ::= `*`
+>
+> ***division-slash*** ::= U+002F `/` SOLIDUS | U+2215 `∕` DIVISION SLASH
+>
+> ***exponent-op*** ::= `^`
+>
+> ***preferential-union*** ::= `&` | U+2294 `⊔` SQUARE CAP
+>
+> ***compare-equal*** ::= `==`
+>
+> ***compare-not-equal*** ::= `!=` | U+2260 `≠` NOT EQUAL TO
+>
+> ***compare-less*** ::= `<`
+>
+> ***compare-less-equal*** ::= `<=` | U+2264 `≤` LESS-THAN OR EQUAL TO
+>
+> ***compare-greater*** ::= `>`
+>
+> ***compare-greater-equal*** ::= `>=` | U+2265 `≥` GREATER-THAN OR EQUAL TO
+>
+> ***left-arrow*** ::= `<-` | U+2190 `←` LEFTWARDS ARROW
+>
+> ***right-arrow*** ::= `->` | U+2192 `→` RIGHTWARDS ARROW
+>
+> ***right-left-arrow*** ::= `<->` | U+21C4 `⇄` RIGHT ARROW OVER LEFT ARROW
+>
+> ***empty-set*** ::= `\0` | U+2205 `∅` EMPTY SET
```

Comment:
Do we have a concrete need for this and `U+221A`?

Reply:
It's used in reaction syntax for an empty complex.